src/solver/pdist/backward_pdist.cpp

```diff
@@ -0,0 +1,284 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <vector>
+
+#include <miopen/miopen.h>
+#include "miopen/conv_solution.hpp"
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/kernel_info.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <miopen/target_properties.hpp>
+#include "miopen/buffer_info.hpp"
+#include "miopen/errors.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/invoke_params.hpp"
+#include "miopen/tensor.hpp"
+#include "miopen/tensor_view_utils.hpp"
+#include <miopen/reduce/utils.hpp>
+
+#include <miopen/pdist.hpp>
+#include <miopen/pdist/solvers.hpp>
+#include <miopen/pdist/invoke_params.hpp>
+#include "miopen/pdist/problem_description.hpp"
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace pdist {
+
+MultiBufferWorkspaceTraits GetMultiBufferWorkspaceTraits(const TensorDescriptor& inputDesc)
+{
+    auto N = inputDesc.GetLengths()[0];
+    auto M = inputDesc.GetLengths()[1];
+
+    auto input_dtype = inputDesc.GetType();
+
+    return MultiBufferWorkspaceTraits{(N - 1) * N * M * get_data_size(input_dtype)};
+}
+
+bool PdistBackward::IsApplicable(const ExecutionContext& context,
+                                 const miopen::pdist::BackwardProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    if(!problem.IsAllContiguous())
+    {
+        return false;
+    }
+
+    if(!problem.IsAllPacked())
+    {
+        return false;
+    }
```

Comment:
this should be in problem description constructor. check https://moreh.atlassian.net/wiki/spaces/MV/pages/903512367/MV-GPU-MIOpen+NetworkConfig+example for more information

Reply:
I'm still a little confused about the `NetworkConfigs` so I want to make clear about this again. Please confirm that my understanding is correct: Given that I've only implemented the contiguous version of `Pdist`, I believe the `IsAllContiguous()` check should be moved into `ProblemDescription`. This would ensure that non-contiguous cases fail at the problem construction stage, as they aren't valid problems under the current implementation.